src/Orleans/Core/Grain.cs

```diff
@@ -308,5 +312,20 @@ protected TGrainState State
         {
             get { return base.GrainState as TGrainState; }
         }
+
+        public Task ClearStateAsync()
```

Comment:
Done with an additional simplification and a added test for the grain interface case.